src/JuliusSweetland.OptiKey/UI/ViewModels/ManagementViewModel.cs

```diff
@@ -112,9 +112,29 @@ private void Ok(Window window)
             }
         }
 
-        private static void Cancel(Window window)
+        private void Cancel(Window window)
         {
-            window.Close();
+            if (ChangesRequireRestart)
```

Comment:
ChangesRequireRestart does not detect if ANY changes have been made, only if changes which require OptiKey to be restarted have been made. Should this not check to see if any changes exist?

Reply:
My mistake, I thought that property represents if any changes have been made in the settings modal view. In this case, I think we would probably just warn users regardless to keep the scope manageable.